pkg/registry/servicecatalog/broker/validation.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
```

Comment:
> I'd like to see the kubelet code. For reference.

Note: this is a jumbo-sized can of worms you're opening, but here you go:

https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet/config

> I think pretty much everything in pkg/registry/servicecatalog should be in pkg/apis/servicecatalog. I don't see any reason for the current split besides convention.

Reasons to keep validations separate from registries:

1.  Making it possible to depend on the API without depending on the guts of the generic registry is highly desirable
2.  You shouldn't have to dig through the REST storage details to figure out what the validations for a type are
3.  Someone might want your API types but not want your registry.

I do also think (like I've said in the past) that producing something that is close to conventional is desired.  Do keep in mind that there are two sides to every coin.  For each departure we make from k8s that people want to do now, there will be a time in the future where new people coming into this project from k8s are going to be asking why we did X or Y.  Those questions take up more time to answer than you might think.  So to me, the choice is between spending time now agreeing as a group to depart from X convention in k8s and then spent time explaining the decision later, or just follow the conventions that aren't terribly stupid (or terribly stupid battle-choosings by the repo champion, ahem) and save the time.

> I agree that validations are closely related to the object definitions, which is why the different strategies have Validate functions. I can fold the validate code directly inline and then validate.go goes away!

Personally I think that it's unwieldy to keep all the validations in `pkg/api/validation`.  I would prefer that we move to a `validation` package for each API group upstream.

How about that for a compromise? `pkg/apis/servicecatalog/validation` holds the validations for our API group; we can make a similar `validation` sub-package in a new API group if we wind up introducing one (I suspect we will for policy objects of some kind eventually).